### PR TITLE
Optionally send json outputs when creating state versions

### PR DESCRIPTION
**Pull request by brandonc**
## Description

Repurposes the json-state field to only include json outputs

## Output from tests
Including output from tests may require access to a TFE instance. Ignore this section if you have no environment to test against.

```
$ TF_ACC=1 go test ./... -v -tags integration -run TestStateVersionsCreate
=== RUN   TestStateVersionsCreate
=== RUN   TestStateVersionsCreate/with_valid_options
=== RUN   TestStateVersionsCreate/with_external_state_representation
=== RUN   TestStateVersionsCreate/with_the_force_flag_set
=== RUN   TestStateVersionsCreate/with_a_run_to_associate_with
    state_version_integration_test.go:245: This can only be tested with the run specific token
=== RUN   TestStateVersionsCreate/without_md5_hash
=== RUN   TestStateVersionsCreate/withous_serial
=== RUN   TestStateVersionsCreate/without_state
=== RUN   TestStateVersionsCreate/with_invalid_workspace_id
--- PASS: TestStateVersionsCreate (14.61s)
    --- PASS: TestStateVersionsCreate/with_valid_options (3.50s)
    --- PASS: TestStateVersionsCreate/with_external_state_representation (2.44s)
    --- PASS: TestStateVersionsCreate/with_the_force_flag_set (3.02s)
    --- SKIP: TestStateVersionsCreate/with_a_run_to_associate_with (0.00s)
    --- PASS: TestStateVersionsCreate/without_md5_hash (0.00s)
    --- PASS: TestStateVersionsCreate/withous_serial (0.00s)
    --- PASS: TestStateVersionsCreate/without_state (0.00s)
    --- PASS: TestStateVersionsCreate/with_invalid_workspace_id (0.00s)
PASS
ok  	github.com/hashicorp/go-tfe	14.869s
?   	github.com/hashicorp/go-tfe/examples/organizations	[no test files]
?   	github.com/hashicorp/go-tfe/examples/workspaces	[no test files]
```

### CODE REVIEW

File: state_version.go [hunk 1]
```diff
@@ -139,13 +139,13 @@ type StateVersionCreateOptions struct {
 	// Optional: Specifies the run to associate the state with.
 	Run *Run `jsonapi:"relation,run,omitempty"`
 
-	// Optional: The external, json representation of state data, base64 encoded.
-	// https://www.terraform.io/internals/json-format#state-representation
-	// Supplying this state representation can provide more details to the platform
+	// Optional: The external, json representation of state outputs, base64 encoded.
+	// https://www.terraform.io/internals/json-format#values-representation (the contents of the
+	// "outputs" attribute) Supplying detailed outputs will provide more details to the platform
```
Comment (annawinkler) [inline state_version.go:144]:
```suggestion
	// "outputs" attribute)
	// Supplying detailed outputs will provide more details to the platform
```

File: state_version_integration_test.go [hunk 2]
```diff
@@ -186,6 +186,8 @@ func TestStateVersionsCreate(t *testing.T) {
 			t.Fatal(err)
 		}
 
+		// TODO: check state outputs for the ones we sent in JSONStateOutputs
```
Comment (annawinkler) [inline state_version_integration_test.go:189]: Can this TODO be resolved?

Comment (brandonc) [reply state_version_integration_test.go:189]: No, not until the API changes are merged.

File: state_version.go [hunk 3]
```diff
@@ -139,13 +139,14 @@ type StateVersionCreateOptions struct {
 	// Optional: Specifies the run to associate the state with.
 	Run *Run `jsonapi:"relation,run,omitempty"`
 
-	// Optional: The external, json representation of state data, base64 encoded.
-	// https://www.terraform.io/internals/json-format#state-representation
-	// Supplying this state representation can provide more details to the platform
+	// Optional: The external, json representation of state outputs, base64 encoded.
+	// https://www.terraform.io/internals/json-format#values-representation (the contents of the
+	// "outputs" attribute)
+	// Supplying detailed outputs will provide more details to the platform
```
Comment (sebasslash) [inline state_version.go:145]: What do you think about rewording this? 
```suggestion
	// Optional: The external, json representation of state outputs, base64 encoded. Supplying this field 
	// will provide more detailed output type information to TFE.
	// For more information on the contents of this field: https://www.terraform.io/internals/json-format#values-representation
```

**Review by annawinkler — DISMISSED** [review]
🚀 Hooray for this PR!

**Pull request merged**